cloud.foundry.cli/src/test/java/cloud/foundry/cli/system/util/SpaceManager.java

```diff
@@ -0,0 +1,321 @@
+     */
+    public SpaceManager(String spaceName) {
+        this.spaceName = spaceName;
+        createSpace();
+    }
+
+    private String safeGetenv(String envVarName) {
+        String value = System.getenv(envVarName);
+
+        if (value == null) {
+            throw new IllegalArgumentException("Environment variable " + envVarName + " is not set");
+        }
+
+        return value;
+    }
+
+    /**
+     * @return CF username (as defined in environment variable)
+     * @throws IllegalArgumentException if environment variable is not defined
+     */
+    public String getCfUsername() {
+        return safeGetenv(CF_USERNAME_ENV_VAR_NAME);
+    }
+
+    /**
+     * @return CF password (as defined in environment variable)
+     * @throws IllegalArgumentException if environment variable is not defined
+     */
+    public String getCfPassword() {
+        return safeGetenv(CF_PASSWORD_ENV_VAR_NAME);
+    }
+
+    /**
+     * @return CF organization (as defined in environment variable)
+     * @throws IllegalArgumentException if environment variable is not defined
+     */
+    public String getCfOrganization() {
+        return safeGetenv(CF_ORGANIZATION_ENV_VAR_NAME);
+    }
+
+    /**
+     * @return CF API endpoint (as defined in environment variable)
+     * @throws IllegalArgumentException if environment variable is not defined
+     */
+    public String getCfApiEndpoint() {
+        return safeGetenv(CF_API_ENDPOINT_ENV_VAR_NAME);
+    }
+
+    /**
+     * @return CF space name
+     */
+    public String getSpaceName() {
+        return spaceName;
+    }
+
+    private void createSpace() {
+        if (servicesOperations != null || applicationsOperations != null) {
+            throw new IllegalStateException("space has been created already");
+        }
+
+        // no need to pollute the method namespace
+        {
+            ArrayList<String> undefinedEnvironmentVariables = new ArrayList<>();
+
+            // query environment variables, remembering all the ones which are not defined
+            // fetching data from the environment is very "cheap", it'd be more annoying to keep them around
+            // it adds a little complexity, but allows us to print them out to help the user set them
+            for (String envVarName : new String[] {
+                    CF_USERNAME_ENV_VAR_NAME,
+                    CF_PASSWORD_ENV_VAR_NAME,
+                    CF_ORGANIZATION_ENV_VAR_NAME,
+                    CF_API_ENDPOINT_ENV_VAR_NAME,
+            }) {
+                String envVarValue = System.getenv(envVarName);
+                if (envVarValue == null) {
+                    undefinedEnvironmentVariables.add(envVarName);
+                }
+            }
+
+            if (!undefinedEnvironmentVariables.isEmpty()) {
+                throw new IllegalStateException("The environment variables " +
+                        Arrays.toString(undefinedEnvironmentVariables.toArray()) + " are not defined");
+            }
```

Comment:
Could be extracted into a helper method. That would increase the readability.

Reply:
It's just called once, though. I've put it in a separate scope for that reason. We could maybe call it everytime we access the environment...